README.md

```diff
@@ -1 +1,260 @@
-# PowerShellTemplate
\ No newline at end of file
+# PowerShellTemplate
+
+This is an example module, following some good practices.
+
+## How to use this Exemplar
+
+### Creating a New Module
+
+To build this module, run `Optimize-Module` on the module directory.
+
+You should then be able to import the module, and run `New-PSModule` which clones the base structure of an example module.
+
+You can use this as follows:
+
+```PowerShell
+New-PSModule -Name Questionmark.Test.Module -Path ~\Git\
+```
+
+Depending on the availability of `dotnet`, this will either use `dotnet new` or perform a naive clone of the directory (replacing all instances of 'CloneModule' with your module name, and editing the PSD1 appropriately).
+
+Alternatively, you can manually clone the folder structure (excluding `.template.config`) from `.\data\CloneModule`.
+
+You will then need to manually edit the PSD1 values, and replace all references to CloneModule with your preferred module name.
+
+### Populating the Module with Functions
+
+`Optimize-Module` creates a single psm1 file from all .ps1 files in folders that aren't specified in CopyDirectories.
+
+We recommend creating a single `FunctionName.ps1` file per function, laid out in public and private folders.
+
+To simplify conversion from a lengthy PowerShell file containing multiple functions to a module, you can call `ConvertFrom-FunctionFile` to create separate files for each function in a given file.
+
+```PowerShell
+ConvertFrom-FunctionFile -Path ~\Git\RefactorProject\SomeScript.ps1 -ModulePath ~\Git\Questionmark.Test.Module
+```
+
+### Classes and Initialization
+
+Classes, enums, and other module prerequisites should be placed in the classes folder. We name these with a digit-prefix (e.g. `00-init.ps1`, `10-ClassDependency.ps1`, `20-SampleClass.ps1`, etc), such that they can be prioritised appropriately and will be added in order at the top of the resultant psm1 file.
```

Comment:
UK/US english conflict: prioritize/Prioritized.

Showing up wrong in my spell check, but might not in yours.